src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1318,11 +1336,246 @@ class Simplex_tree {
+     * Nodes with label @p u get affected only if a Node with label @p v is in their same
+     * siblings set.
+     * We then try to insert "ponctually" @p v all over the subtree rooted
+     * at `Node(u)`. Each insertion of a Node with @p v label induces a local
+     * expansion at this Node (as explained above) and a sequence of "ponctual"
+     * insertion of `Node(v)` in the subtree rooted at sibling nodes of the new node,
+     * on its left.
+     */
+
+    static_assert(Options::link_nodes_by_label, "Options::link_nodes_by_label must be true");
+
+    if (u == v) { // Are we inserting a vertex?
+      auto res_ins = root_.members().emplace(u, Node(&root_,fil));
+      if (res_ins.second) { //if the vertex is not in the complex, insert it
+        added_simplices.push_back(res_ins.first); //no more insert in root_.members()
+        update_simplex_tree_after_node_insertion(res_ins.first);
+      }
+      return; //because the vertex is isolated, no more insertions.
+    }
+    // else, we are inserting an edge: ensure that u < v
+    if (v < u) { std::swap(u,v); }
+
+    //Note that we copy Simplex_handle (aka map iterators) in added_simplices
+    //while we are still modifying the Simplex_tree. Insertions in siblings may
+    //invalidate Simplex_handles; we take care of this fact by first doing all
+    //insertion in a Sibling, then inserting all handles in added_simplices.
+
+    //check whether vertices u and v are in the tree. If not, return an error.
+    auto sh_u = root_.members().find(u);
+    GUDHI_CHECK(sh_u != root_.members().end() &&
+          root_.members().find(v) != root_.members().end(),
+          std::invalid_argument(
+                  "Simplex_tree::insert_edge_as_flag - insert an edge whose vertices are not in the complex")
+                );
+
+    //check if the edge {u,v} is already in the complex, if true, nothing to do.
+    if (has_children(sh_u) &&
+        sh_u->second.children()->members().find(v) != sh_u->second.children()->members().end())
+    {
+      return;
+    }
+
+    // //upper bound on dimension
+    // //TODO: avoid setting dimension_to_be_lowered_ to true by testing dimension of added simplices.
+    if (dimension_ < dim_max) {
+      dimension_ = dim_max;
+      dimension_to_be_lowered_ = true;
+    } else if (dim_max == -1 && !dimension_to_be_lowered_) {
+      ++dimension_;
+      dimension_to_be_lowered_ = true;
+    }
+    // const auto tmp_dim = dimension_;
+    // auto tmp_max_dim = dimension_;
+
+    //for all siblings containing a Node labeled with u (including the root), run
+    //compute_punctual_expansion
+    //todo parallelise
+    List_max_vertex* nodes_with_label_u = nodes_by_label(u);//all Nodes with u label
+
+    GUDHI_CHECK(nodes_with_label_u != nullptr,
+                "Simplex_tree::insert_edge_as_flag - cannot find the list of Nodes with label u");
+
+    for (auto hook_u_it = nodes_with_label_u->begin(); hook_u_it != nodes_with_label_u->end(); ++hook_u_it)
```

Comment:
Did you roll back the syntax because it was copying? I think it should have been `for (auto&& node_as_hook : *nodes_with_label_u)` (or just a single `&` if you prefer).

Reply:
I didn't even saw that there were no `&`, no wonder it was not working. Thanks!